### ISSUE TITLE
False positive for `function-whitespace-after` in a mixin

### ISSUE THREAD

**Issue by onigoetz** [stylelint/stylelint#1370]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

When I use variable interpolation inside a mixin, `function-whitespace-after` returns a false positive.

Removing the parenthesis in `$(att)` breaks the feature. so that's not an option

> Which rule, if any, is this issue related to?

`function-whitespace-after`

> What CSS is needed to reproduce this issue?

```css
@define-mixin property-px2vw $att, $px {
    $(att): calc(1px * $px);
    $(att): calc(1vw * ($px/16));
}

.class {
	@mixin property-px2vw prop, 20px;
}
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "rules": {
     "function-whitespace-after": "always"
  }
}
```

> Which version of stylelint are you using?

`5.2.1`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

postcss plugin

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Yes, postcss mixins with the `postcss-mixins` plugin (https://github.com/postcss/postcss-mixins)

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors you are getting)?

The following warnings were flagged:

```shell
test.css
2:11	⚠  Expected whitespace after ")" (function-whitespace-after) [stylelint]
3:11	⚠  Expected whitespace after ")" (function-whitespace-after) [stylelint]
```

**Comment by davidtheclark** [comment]
This fix might be as easy as adding `:` to the list of characters that are ok after the closing parenthesis at https://github.com/stylelint/stylelint/blob/master/src/rules/function-whitespace-after/index.js#L43.

**Comment by davidtheclark** [comment]
Oh wait ... with postcss-simple-vars interpolation there could be a closing parentheses *anywhere*, e.g. `some$(thing)here`. So maybe we need to add an exception that say if the function name is `$` we ignore it — it's not a standard function. Probably this rule should use the newish `isStandardFunction` util, and that util should make sure to account for postcss-simple-vars.

**Comment by pikilon** [comment]
the problem occurs too using postcss variables like this:

```css
@define-mixin property-px2vw $att, $px {
    var(--att): calc(1px * $px);
    var(--att): calc(1vw * ($px/16));
}
```
So there is no workaround.

**Comment by davidtheclark** [comment]
Didn't know you could do that with `var()` functions. I know you can't *interpolate* those, so this should be addressed if we do all of the above: add `:` to the list of characters that are acceptable after closing parenthesis, and register as non-standard functions whose name is `$`.

> So there is no workaround.

Not sure what this means. If you're looking for a way to stop getting warnings, you could always insert a space after the closing parenthesis. Or turn the rule off.

**Comment by pikilon** [comment]
> If you're looking for a way to stop getting warnings, you could always insert a space after the closing parenthesis. Or turn the rule off.

The company I work for has this linter to commit/push any css, so the only thing I can do is compile scss and restyle the result.

**Comment by onigoetz** [comment]
Thank you very much for the fix, do you have an Idea when it will be released ?

I see it's in the v7 branch

**Comment by jeddy3** [comment]
We hope to get `6.6` out in the next day or two. We then intend to focus on `v7`. There's a [small backlog of issues](https://github.com/stylelint/stylelint/milestones/7.0.0) for `v7` though, so any help knocking these off will help get the `v7` release out sooner.
